### PR TITLE
[HOTFIX] Fix issue: Custom tests that do not have MLIR_ENABLED are disabled by default

### PR DESCRIPTION
**Pull request by atamazov**
The bug is originated from my PR #1246 (which is unsuccessful attempt to fix another problem with MLIR testing). Currently many custom tests (including new test_regression_half_mi200) are not working.

## :warning: In order to prevent leaking regressions into develop, it is highly recommended to pause merging other PRs until this one is merged (and other PRs merge from `develop` after that and pass all tests).

### CODE REVIEW

**Comment by junliume** [comment]
> The bug is originated from my PR #1246 (which is unsuccessful attempt to fix another problem with MLIR testing). Currently many custom tests (including new test_regression_half_mi200) are not working.
> 
> ## ⚠️ In order to prevent leaking regressions into develop, it is highly recommended to pause merging other PRs until this one is merged (and other PRs merge from `develop` after that and pass all tests).

I can pause some other PRs to yield CI resources for #1271 and #1278 
Anything we can do to process the CI selectively? The whole run might take much longer. Thanks!

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -651,14 +648,14 @@ set(IMPLICITGEMM_MLIR_ARGS_F ${IMPLICITGEMM_ARGS} --verbose --disable-backward-d
 set(IMPLICITGEMM_MLIR_ARGS_B ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-weights)
 set(IMPLICITGEMM_MLIR_ARGS_W ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-data)
 
-add_custom_test(test_conv_igemm_mlir_small HALF_ENABLED MLIR_ENABLED GFX908_DISABLED GFX90A_DISABLED
+add_custom_test(test_conv_igemm_mlir_small HALF_ENABLED SKIP_UNLESS_MLIR GFX908_DISABLED GFX90A_DISABLED
```
Comment (jerryyin) [inline test/CMakeLists.txt:651]: Why is this necessary? What was the problem in the implementation of `MLIR_ENABLED`?

This seems as if we'd still need to keep the MLIR smoke tests after #1228, is that right?

Comment (atamazov) [reply test/CMakeLists.txt:651]: > Why is this necessary? What was the problem in the implementation of MLIR_ENABLED?

Due to a logic error in computation of `is_mlir_check`, currently `is_mlir_check` is always false, unless both MLIR_ENABLED attribute and MIOPEN_TEST_MLIR are set. This leads to that all custom tests are disabled by default.

Comment (atamazov) [reply test/CMakeLists.txt:651]: > This seems as if we'd still need to keep the MLIR smoke tests after #1228, is that right?

This is out of scope of this PR, but yes, because MLIR tests are intended to test MLIR solvers. If, due to some reason, we will disable MLIR, then these tests should be skipped. This applies to all MLIR tests, not only smoke ones.

**Comment by atamazov** [comment]
@junliume 
> Anything we can do to process the CI selectively? The whole run might take much longer. 

No, we should run all tests to make sure that there is no new failures (that may appear due to re-enabled custom tests).

**Comment by atamazov** [comment]
:warning: New failure: Newly enabled `test_conv_group` is failing for Fp32 Hip gfx908a. I am working on this.

File: test/CMakeLists.txt (hunk 1 above)
Comment (jerryyin) [reply test/CMakeLists.txt:651]: Sounds good, I will amend the jenkinsfile once your PR is merged.

**Comment by junliume** [comment]
@atamazov CI pipeline `#3` and `#4` are ongoing simultaneously for `be086ff`. I think we only need to test **Full Tests II** and after, right?

**Comment by atamazov** [comment]
#3 can be terminated, but in this case, YES, we need to run at least Full Tests II for all targets. #4 is for quick verification on gfx90a.

**Comment by junliume** [comment]
> #3 can be terminated, but in this case, YES, we need to run at least Full Tests II for all targets. #4 is for quick verification on gfx90a.

Done, and a minor issue is that we cannot split Full Tests I and Full Tests II (easily) via the build parameters?
Monitoring run `#4` and `#5`.

**Comment by atamazov** [comment]
@junliume 
> Done, and a minor issue is that we cannot split Full Tests I and Full Tests II (easily) via the build parameters?

In this case it is possible to filter out all data types except FP32.

**Comment by atamazov** [comment]
Full job `#9` (except static & package) launched the double check that everything is fine.

**Review by junliume — APPROVED** [review]
We have ensured that test_conv_group is passing.

Run `#10` is running for safety check but we can leave it as post-merge.
